framework/include/vectorpostprocessors/ElementVariablesDifferenceMax.h

```diff
@@ -0,0 +1,75 @@
+#ifndef ELEMENTVARIABLESDIFFERENCEMAX_H
+#define ELEMENTVARIABLESDIFFERENCEMAX_H
+
+#include "ElementVectorPostprocessor.h"
+
+// Forward Declarations
+class ElementVariablesDifferenceMax;
+
+template <>
+InputParameters validParams<ElementVariablesDifferenceMax>();
+
+/**
+ * \brief     Finds the largest difference between two variable fields
+ *
+ * \details   The location of the associated quadrature point is also stored,
+ *            along with the actual field values that resulted in the
+ *            difference. The user may specify in the input file whether to find
+ *            the largest absolute difference and, if so, whether to preserve
+ *            the sign
+ */
+class ElementVariablesDifferenceMax : public ElementVectorPostprocessor
+{
+public:
+  // Basic MooseObject cosntructor
+  ElementVariablesDifferenceMax(const InputParameters & parameters);
+
+  // Required overrides
+  virtual void execute() override;
+  virtual void finalize() override;
+  virtual void initialize() override;
+  virtual void threadJoin(const UserObject & s) override;
+
+protected:
+  /// The first variable, operated to produce a difference as: #_a - #_b
+  const VariableValue & _a;
+
+  /// The second variable, operated to produce a difference as: #_a - #_b
+  const VariableValue & _b;
+
+  /// The value of #_a that produced the maximum difference
+  VectorPostprocessorValue & _a_value;
+
+  /// The value of #_b that produced the maximum difference
+  VectorPostprocessorValue & _b_value;
+
+  /// The actual maximum difference
+  VectorPostprocessorValue & _max_difference;
+
+  /// The x position of the associated quadrature point
+  VectorPostprocessorValue & _position_x;
+
+  /// The y position of the associated quadrature point
+  VectorPostprocessorValue & _position_y;
+
+  /// The z position of the associated quadrature point
+  VectorPostprocessorValue & _position_z;
+
+  /**
+   * Indicates whether #_max_difference is actually negative, used only in
+   * finalize() if #_preserve_sign is set to true
+   */
+  bool _max_difference_is_negative;
+
+private:
```

Comment:
I don;t see a compelling reason to make the members below private. Protected is just fine.